VideoLocker/src/main/java/org/edx/mobile/view/UserProfileFragment.java

```diff
@@ -168,26 +171,25 @@ private void setAccount(@Nullable final Account account) {
                 viewHolder.locationContainer.setVisibility(View.GONE);
             } else {
                 viewHolder.limitedView.setVisibility(View.GONE);
-                if (account.getLanguageProficiencies().isEmpty()) {
-                    viewHolder.languageContainer.setVisibility(View.GONE);
-                } else {
-                    viewHolder.languageContainer.setVisibility(View.VISIBLE);
-                    viewHolder.languageText.setText(
-                            new Locale.Builder()
-                                    .setLanguage(account.getLanguageProficiencies().get(0).getCode())
-                                    .build()
-                                    .getDisplayName());
+                viewHolder.languageContainer.setVisibility(View.GONE);
+                if (!account.getLanguageProficiencies().isEmpty()) {
+                    try {
+                        viewHolder.languageText.setText(
+                                LocaleUtils.getLanguageNameFromCode(account.getLanguageProficiencies().get(0).getCode()));
```

Comment:
@miankhalid: This is discussed in the comment at https://openedx.atlassian.net/browse/MA-1283?focusedCommentId=145399&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-145399